scripts/obtain/uninstall/dotnet-uninstall-pkgs.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+current_user=$(whoami)
+if [ $current_user != "root" ]; then
+    echo "`basename "$0"` uninstallation script requires superuser privileges to run"
+    exit 1
+fi
+
+# this is the common suffix for all the dotnet pkgs
+dotnet_pkg_name_suffix="com.microsoft.dotnet"
+dotnet_install_root="/usr/local/share/dotnet"
+dotnet_path_file="/etc/paths.d/dotnet"
+
+remove_all(){
+    installed_pkgs=($(pkgutil --pkgs | grep $dotnet_pkg_name_suffix))
+    
+    for i in "${installed_pkgs[@]}"
+    do
+        echo "Removing dotnet component - \"$i\""
+        pkgutil --force --forget "$i"
+    done
+    
+    echo "Deleting install root - $dotnet_install_root"
+    rm -rf $dotnet_install_location
+    rm -f $dotnet_path_file
+}
+
+remove_all
+[ "$?" -ne 0 ] && echo "Failed to remove dotnet packages." && exit 1
```

Comment:
Have you tested a case where an error occurs?

Without `set -e` or some test at the end of the function `ie. [ -z "$out" ]` I believe the exit code of the function will only be that of the last command run `rm -f` which I wouldn't ever expect to return non-zero.